src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,322 @@
+      true,
+    )
+    const _lastTabbableChild = getLastTabbable(
+      this._root.current,
+      this._root.current.lastChild as HTMLElement,
+      true,
+    )
+
+    if (ev.shiftKey && _firstTabbableChild === ev.target) {
+      focusAsync(_lastTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    } else if (!ev.shiftKey && _lastTabbableChild === ev.target) {
+      focusAsync(_firstTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    }
+  }
+
+  private _forceFocusInTrap = (ev: FocusEvent): void => {
+    if (
+      FocusTrapZone._focusStack.length &&
+      this === FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+    ) {
+      const focusedElement = document.activeElement as HTMLElement
+
+      if (!this._root.current.contains(focusedElement)) {
+        this.focus()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _forceClickInTrap = (ev: MouseEvent): void => {
+    if (
+      FocusTrapZone._focusStack.length &&
+      this === FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+    ) {
+      const clickedElement = ev.target as HTMLElement
+
+      if (clickedElement && !this._root.current.contains(clickedElement)) {
+        this.focus()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _subscribeToEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.sub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.sub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _unsubscribeFromEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.unsub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.unsub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _getPreviouslyFocusedElementOutsideTrapZone = () => {
+    const { elementToFocusOnDismiss } = this.props
+    let previouslyFocusedElement = this._previouslyFocusedElementOutsideTrapZone
+
+    if (elementToFocusOnDismiss && previouslyFocusedElement !== elementToFocusOnDismiss) {
+      previouslyFocusedElement = elementToFocusOnDismiss
+    } else if (!previouslyFocusedElement) {
+      previouslyFocusedElement = document.activeElement as HTMLElement
+    }
+
+    return previouslyFocusedElement
+  }
+
+  private _hideContentFromAccessibilityTree = () => {
+    const elements = (document.body && document.body.children) || []
```

Comment:
it would be better to name this variable as `bodyChildren` - this would make the intent of this method easier to grasp

Reply:
agree